lib/core/utils/contains.js

```diff
@@ -7,33 +7,25 @@
  * @return {Boolean}           Whether `vNode` contains `otherVNode`
  */
 export default function contains(vNode, otherVNode) {
-  /*eslint no-bitwise: 0*/
-  if (vNode.shadowId || otherVNode.shadowId) {
-    do {
-      if (vNode.shadowId === otherVNode.shadowId) {
-        return true;
```

Comment:
This was wrong. We never caught it because `contains` is never called with vNode as a node in the shadow DOM tree.